File: scripts/commands/generic/cmdPVP.js
var execute = function(galManager, player, target, args) 
{
	var creature = player.getCreatureObject();
	var PvpFaction = Java.type("resources.PvpFaction");
	var FactionIntent = Java.type("intents.FactionIntent");
	var FactionIntentType = Java.type("intents.FactionIntent.FactionIntentType");
	var ChatSystemMessage = Java.type("network.packets.swg.zone.chat.ChatSystemMessage");
	var SystemChatType = Java.type("network.packets.swg.zone.chat.ChatSystemMessage.SystemChatType");
	var intent = null;
	
	// Ziggy was using this to test until recruiters are enabled.

	if (args.length > 0) {
		if (args.indexOf("imperial") > -1) {
			intent = new FactionIntent(creature, PvpFaction.IMPERIAL);
		}
		else if (args.indexOf("rebel") > -1) {
			intent = new FactionIntent(creature, PvpFaction.REBEL);
		}
		else {
			intent = new FactionIntent(creature, PvpFaction.NEUTRAL);
		}
		intent.broadcast();
	} else if (creature.getPvpFaction() != PvpFaction.NEUTRAL) {
		intent = new FactionIntent(creature, FactionIntentType.STATUSUPDATE);
		intent.broadcast();
	} else {
		creature.getOwner().sendPacket(new ChatSystemMessage(SystemChatType.SCREEN_AND_CHAT, "@faction_recruiter:not_aligned"));
	}
		
};

File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/combat/CmdPVP.java
package com.projectswg.holocore.resources.support.global.commands.callbacks.combat;

import com.projectswg.common.data.encodables.tangible.PvpFaction;
import com.projectswg.holocore.intents.gameplay.gcw.faction.FactionIntent;
import com.projectswg.holocore.intents.support.global.chat.SystemMessageIntent;
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback;
import com.projectswg.holocore.resources.support.global.player.Player;
import com.projectswg.holocore.resources.support.objects.swg.SWGObject;
import com.projectswg.holocore.resources.support.objects.swg.creature.CreatureObject;
import org.jetbrains.annotations.NotNull;

public final class CmdPVP implements ICmdCallback {
	
	@Override
	public void execute(@NotNull Player player, SWGObject target, @NotNull String args) {
		CreatureObject creature = player.getCreatureObject();
		
		if (creature.getPvpFaction() != PvpFaction.NEUTRAL) {
			new FactionIntent(creature, FactionIntent.FactionIntentType.SWITCHUPDATE).broadcast();
		} else {
			SystemMessageIntent.broadcastPersonal(player, "@faction_recruiter:not_aligned");
		}
	}
	
}
